src/sif/autocost.cc

```diff
@@ -926,9 +933,11 @@ class TaxiCost : public AutoCost {
                         const graph_tile_ptr& tile,
                         const baldr::TimeInfo& time_info,
                         uint8_t& flow_sources) const override {
-    auto edge_speed = tile->GetSpeed(edge, flow_mask_, time_info.second_of_week, false, &flow_sources,
-                                     time_info.seconds_from_now);
-    auto final_speed = std::min(edge_speed, top_speed_);
+    auto edge_speed = fixed_speed_ == 0
+                          ? tile->GetSpeed(edge, flow_mask_, time_info.second_of_week, false,
+                                           &flow_sources, time_info.seconds_from_now)
+                          : fixed_speed_;
+    auto final_speed = fixed_speed_ != 0 ? fixed_speed_ : std::min(edge_speed, top_speed_);
```

Comment:
edge_speed is already fixed_speed from above
```suggestion
    auto final_speed = std::min(edge_speed, top_speed_);
```

Reply:
same is in the other costing models

Reply:
ah never mind this comment.. of course you want to fix the final speed.